Wikipedia/Code/NavigationStateController.swift

```diff
@@ -8,6 +9,7 @@ protocol DetailPresentingFromContentGroup {
 final class NavigationStateController: NSObject {
     private let dataStore: MWKDataStore
     private var theme = Theme.standard
+    private var settingsNavController: UINavigationController?
```

Comment:
@mazevedofs One last minor thing - add `weak` here -

`private weak var settingsNavController: UINavigationController?`

Unless you disagree of course. Thanks!

Reply:
I totally agree, completely missed it from your previous comment